MVVM-C/ViewModel/ArticleViewModel.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  ArticleViewModel.swift
+//  MVVM-C
+//
+//  Created by Usemobile on 09/06/20.
+//  Copyright © 2020 Jakub Homik. All rights reserved.
+//
+
+import Foundation
+
+// ViewModel: Its responsible for holding the model and formatting it to be displayed to UI component, holding state
+
+/*
+ Model                     <---> ViewModel <---> UI
+ (Article, ArticleManager) <---> ViewModel <---> (GotCell)
+ */
+
+final class ArticleViewModel {
+    
+    private let model: Article
+    
+    typealias FavoriteObserver = ((Bool) -> Void)
+    var onFavorite = [NSObject: FavoriteObserver]()
+    
+    var title: String {
+        return self.model.title
+    }
+    var abstract: String {
+        return self.model.abstract
+    }
+    var thumbnail: String? {
+        return self.model.thumbnail
+    }
+    var url: String {
+        return self.model.url
+    }
+    
+    private(set) var isFavorited: Bool {
+        didSet {
+            self.onFavorite.keys.forEach { key in
+                self.onFavorite[key]?(self.isFavorited)
+            }
+        }
+    }
```

Comment:
this property isFavorited is really confusing me. How this working ?

Reply:
This property is used to store if the Article has been marked as a favorite or not, maybe the I could rename it to give more context. In that case I want that only my ViewModel be able to update its value, and any object to be able to read it, so I've set the `private(set)` modifier, which states that this property can only be set at its declaration class scope. That's just an way I want it to work, to centralize that property control.